libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/offroute/OffRouteDetector.java

```diff
@@ -226,24 +224,28 @@ private static boolean movingAwayFromManeuver(RouteProgress routeProgress,
     LineString remainingStepLineString = TurfMisc.lineSlice(userPointOnStep, maneuverPoint, stepLineString);
     double userDistanceToManeuver = TurfMeasurement.length(remainingStepLineString, TurfConstants.UNIT_METERS);
 
-    boolean hasDistances = !distancesAwayFromManeuver.isEmpty();
-    boolean validOffRouteDistanceTraveled = hasDistances && distancesAwayFromManeuver.peekLast()
-      - distancesAwayFromManeuver.peekFirst() < MINIMUM_BACKUP_DISTANCE_FOR_OFF_ROUTE;
-    boolean exceedsManeuverDistancesThreshold = validOffRouteDistanceTraveled
-      && distancesAwayFromManeuver.size() >= 3;
-
-    if (exceedsManeuverDistancesThreshold) {
-      // User's moving away from maneuver position, thus offRoute.
-      return true;
-    }
     if (distancesAwayFromManeuver.isEmpty()) {
-      distancesAwayFromManeuver.push((int) userDistanceToManeuver);
-    } else if (userDistanceToManeuver > distancesAwayFromManeuver.peek()) {
-      distancesAwayFromManeuver.push((int) userDistanceToManeuver);
-    } else {
-      // If we get a descending distance, reset the counter
+      // No move-away positions before, add the current one to history stack
+      distancesAwayFromManeuver.add((int) userDistanceToManeuver);
+    } else if ((int) userDistanceToManeuver > distancesAwayFromManeuver.getLast()) {
+      // If distance to maneuver increased (wrong way), add new position to history stack
+
+      if (distancesAwayFromManeuver.size() >= 3) {
+        // We replace the first one, in order to keep the history with the last one
+        distancesAwayFromManeuver.removeLast();
+      }
+      distancesAwayFromManeuver.add((int) userDistanceToManeuver);
```

Comment:
Since `add` is the same as `addLast`, should we use addLast instead, as it makes it clearer what is happening?